internal/provider/kubernetes/predicates.go

```diff
@@ -42,6 +42,48 @@ func (r *gatewayAPIReconciler) hasMatchingController(obj client.Object) bool {
 	return false
 }
 
+// hasMatchingNamespaceLabels returns true if the namespace of provided object has
+// the provided labels or false otherwise.
+func (r *gatewayAPIReconciler) hasMatchingNamespaceLabels(labels []string) func(client.Object) bool {
+	return func(obj client.Object) bool {
+		nsString := obj.GetNamespace()
+		ns := &corev1.Namespace{}
+		if err := r.client.Get(
+			context.Background(),
+			client.ObjectKey{
+				Namespace: "", // Namespace object should have empty Namesapce
+				Name:      nsString,
+			},
+			ns,
+		); err != nil {
+			// Question: what's the common practice to handle error in predicate?
```

Comment:
Could anyone help me to understand the common practices in Enovy Gateway to handle errors in predicate?

Reply:
logging an error and returning `false` is fine